client/src/Components/Card/DiagnosticCard.js

```diff
@@ -2,22 +2,43 @@ import React, { Component } from 'react';
 import { Card, Icon } from 'semantic-ui-react';
 
 class DiagnosticCard extends Component {
-  render() {
-    return (
-      <Card className="data-cards-r1">
-        <Card.Content>
-          <Icon circular inverted color="red" name="wrench" size="big" />
-          <div className="data-content">
-            <p>Diagnostic Issues</p>
-            <h2>{this.props.diagnosticIssue}</h2>
-          </div>
-          <Card.Content extra>
-            <hr></hr>
-            <a href="/#">{this.props.diagnosticDetail}</a>
+  render(props) {
+    if (
+      this.props.diagnosticIssue[0] === undefined ||
```

Comment:
this.props.diagnosticIssue && this.props.diagnosticIssue.length > 0

Reply:
This still crashes

Reply:
what is the error? Did you definitely change it to ```this.props.diagnosticIssue && this.props.diagnosticIssue.length > 0```?

Reply:
TypeError: this.props.diagnosticIssue[0] is undefined is the error and it won't display error codes from cars that do have error codes, i played around with it a bit

Reply:
I have no doubt it's problem with the way i am displaying the data on the <h2>

Reply:
have you tried putting a debugger statement in to see what is being passed in as props?